### ISSUE TITLE
Suggestions are not working in Windows10 + WSL2

### ISSUE THREAD

**Issue by kaviththiranga** [ballerina-platform/ballerina-lang#37092]
**Description:**
Module level and block level suggestions are not working. Suggestions are only coming when I type an import alias(eg: ```io:```)

![image (1)](https://user-images.githubusercontent.com/1505855/180147555-22600945-3a35-4e0d-b7d8-c5e51840a30a.png)
![image](https://user-images.githubusercontent.com/1505855/180147559-464ff9b0-3d13-4127-a91f-394afd49da16.png)

![image (2)](https://user-images.githubusercontent.com/1505855/180147548-d27f8095-1700-4eb6-8a57-23e8676ff402.png)

But for the same project and versions, it works in Mac OSX.

<img width="1008" alt="Screenshot 2022-07-21 at 12 01 48" src="https://user-images.githubusercontent.com/1505855/180147702-56479738-e7ba-42e3-a4d1-4efdfb626724.png">

**Steps to reproduce:**

**Affected Versions:**
Ballerina 2201.1.1 (Swan Lake Update 1)
Language specification 2022R2
Update Tool 1.3.9
VSCode Extension 3.0.2

**OS, DB, other environment details and versions:**
Windows 10 and WSL2 (Ubuntu)

**Related Issues (optional):**

**Suggested Labels (optional):**

Priority/Blocker
Team/LanguageServer

**Suggested Assignees (optional):**

**Comment by IMS94** [comment]
@kaviththiranga Will it be possible to get the trace and debug logs of the vscode plugin when this is happening?

**Comment by kaviththiranga** [comment]
@IMS94 sure, will share before EOD, need to switch to my Desktop computer to reproduce this.

**Comment by kaviththiranga** [comment]
@IMS94 I see below error in debug logs.

```
[Info  - 6:45:46 AM] Operation 'text/didChange' {fileUri: 'file:///home/kavith/balprojects/testActions/main.bal'} updated
[Error - 6:45:47 AM] Operation 'text/completion' failed! {uri: '/home/kavith/balprojects/testActions/main.bal', [18:1], error: 'Invalid version: '1.3.1_temp'. Unexpected character 'ILLEGAL(_)' at position '5', expecting '[HYPHEN, PLUS, EOI]''}
io.ballerina.projects.ProjectException: Invalid version: '1.3.1_temp'. Unexpected character 'ILLEGAL(_)' at position '5', expecting '[HYPHEN, PLUS, EOI]'
	at io.ballerina.projects.SemanticVersion.from(SemanticVersion.java:45)
	at io.ballerina.projects.PackageVersion.from(PackageVersion.java:44)
	at org.ballerinalang.langserver.LSPackageLoader.lambda$getPackagesFromRepository$4(LSPackageLoader.java:165)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.ballerinalang.langserver.LSPackageLoader.lambda$getPackagesFromRepository$5(LSPackageLoader.java:156)
	at java.base/java.util.HashMap.forEach(HashMap.java:1336)
	at org.ballerinalang.langserver.LSPackageLoader.getPackagesFromRepository(LSPackageLoader.java:152)
	at org.ballerinalang.langserver.LSPackageLoader.getRemoteRepoPackages(LSPackageLoader.java:92)
	at org.ballerinalang.langserver.LSPackageLoader.getPackagesFromBallerinaUserHome(LSPackageLoader.java:144)
	at org.ballerinalang.langserver.LSPackageLoader.getAllVisiblePackages(LSPackageLoader.java:123)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.getModuleCompletionItems(AbstractCompletionProvider.java:362)
	at org.ballerinalang.langserver.completions.providers.AbstractCompletionProvider.getTypeDescContextItems(AbstractCompletionProvider.java:318)
	at org.ballerinalang.langserver.completions.providers.context.ModulePartNodeContext.getModulePartContextItems(ModulePartNodeContext.java:162)
	at org.ballerinalang.langserver.completions.providers.context.ModulePartNodeContext.getCompletions(ModulePartNodeContext.java:89)
	at org.ballerinalang.langserver.completions.providers.context.ModulePartNodeContext.getCompletions(ModulePartNodeContext.java:49)
	at org.ballerinalang.langserver.completions.util.CompletionUtil.route(CompletionUtil.java:125)
	at org.ballerinalang.langserver.completions.util.CompletionUtil.getCompletionItems(CompletionUtil.java:81)
	at org.ballerinalang.langserver.completions.BallerinaCompletionExtension.execute(BallerinaCompletionExtension.java:65)
	at org.ballerinalang.langserver.completions.BallerinaCompletionExtension.execute(BallerinaCompletionExtension.java:40)
	at org.ballerinalang.langserver.LangExtensionDelegator.completion(LangExtensionDelegator.java:122)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$completion$0(BallerinaTextDocumentService.java:147)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

[Info  - 6:46:18 AM] Operation 'text/didOpen' {fileUri: 'file:///home/kavith/balprojects/testActions/Ballerina.toml'} opened
```

**Comment by kaviththiranga** [comment]
![image](https://user-images.githubusercontent.com/1505855/182059215-2eb8cad1-7069-4251-ad57-8cd0d2fc96d9.png)

![image](https://user-images.githubusercontent.com/1505855/182059303-5fb7dd62-546d-4c0d-ab05-4474a752a166.png)

**Comment by kaviththiranga** [comment]
```toml
[package]
org = "kavith"
name = "testActions"
version = "0.1.0"
distribution = "2201.1.1"

[build-options]
observabilityIncluded = true

```

**Comment by IMS94** [comment]
@kaviththiranga thanks! This happens due to the temp folders left unclean when a pull package failed. Note the `_temp` suffix in the version being printed. The LSPackageLoader which got affected is improved in #37181.

And the original cause where `bal pull` leave corrupt bala files is addressed in #34939

**Comment by azinneera** [comment]
Duplicate of https://github.com/ballerina-platform/ballerina-lang/issues/36470

We fixed this for Update 3. We can add the same fix to Update 1 next patch and Update 2.

**Comment by IMS94** [comment]
@azinneera It's better if we can add the same fix to 2201.1.x next patch and 2201.2.0. Because in Choreo they used to mound the ballerina home to avoid pulling packages again (not sure if this design has changed). Therefore, there can be workspaces which has invalid bala files.

**Comment by azinneera** [comment]
> @azinneera It's better if we can add the same fix to 2201.1.x next patch and 2201.2.0. Because in Choreo they used to mound the ballerina home to avoid pulling packages again (not sure if this design has changed). Therefore, there can be workspaces which has invalid bala files.

We merged the fix to 2201.1.x too.

**Comment by malinthar** [comment]
Closing the issue as this was fixed with https://github.com/ballerina-platform/ballerina-lang/pull/37181
